fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingMerger.java

```diff
@@ -117,6 +119,7 @@ private CompletableFuture<Boolean> handleSuccess(final IndexDeferredMaintenanceC
         if (repartitionDocumentCount > 0) {
             repartitionDocumentCount = 0;
         }
+        repartitionCapped = mergeControl.repartitionCapped();
```

Comment:
It feels that this `repartitionCapped` should not be a member. Will you consider providing `mergeControl` as an argument to `shouldGiveRepartitionSecondChance`, and let that function keep the repartitioning knowledge?